### PR TITLE
Add unit tests to jobset success policy functions

### LINKED ISSUE #491: Add unit tests to jobset success policy functions
**What would you like to be added**:

Add unit tests to jobset success policy functions [here](https://github.com/kubernetes-sigs/jobset/blob/de528b8c86e39a094e430500be277f1e9ec6c409/pkg/controllers/success_policy.go#L24)

**Why is this needed**:
To improve test coverage

### PR DESCRIPTION
**Pull request by zhifei92**
my first pr

Add unit tests to jobset success policy functions

Fixes #491

### CODE REVIEW

**Comment by googs1025** [comment]
hi, Is this a duplicate pull request? https://github.com/kubernetes-sigs/jobset/pull/493

**Comment by danielvegamyhre** [comment]
Please use table-driven format to test different scenarios for a given function. Example: https://github.com/kubernetes-sigs/jobset/blob/8860f832dee2c35d9cb2951260208fcef87cd2db/pkg/controllers/startup_policy_test.go#L25-L45

**Comment by zhifei92** [comment]
sorry, I didn't notice，

> Please use table-driven format to test different scenarios for a given function. Example:
> 
> https://github.com/kubernetes-sigs/jobset/blob/8860f832dee2c35d9cb2951260208fcef87cd2db/pkg/controllers/startup_policy_test.go#L25-L45

get

**Comment by zhifei92** [comment]
> hi, Is this a duplicate pull request? #493

Sorry, I didn't notice

**Comment by zhifei92** [comment]
@danielvegamyhre @kannon92 hello, please help me review，thx

File: pkg/controllers/success_policy_test.go [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+)
+
+func TestJobMatchesSuccessPolicy(t *testing.T) {
+	var (
+		jobSetName = "test-jobset"
+		jobName    = "test-job"
+		ns         = "default"
+	)
+
+	jobMatchesSuccessPolicyTestCases := []struct {
```
Comment (dejanzele) [inline pkg/controllers/success_policy_test.go:33]:
```suggestion
	tests := []struct {
```

File: pkg/controllers/success_policy_test.go [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+	}
+}
+
+func TestReplicatedJobMatchesSuccessPolicy(t *testing.T) {
+	var (
+		jobSetName = "test-jobset"
+		ns         = "default"
+	)
+
+	replicatedJobMatchTestCases := []struct {
```
Comment (dejanzele) [inline pkg/controllers/success_policy_test.go:90]:
```suggestion
	tests := []struct {
```

File: pkg/controllers/success_policy_test.go [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+}
+
+func TestNumJobsMatchingSuccessPolicy(t *testing.T) {
+	var (
+		jobSetName = "test-jobset"
+		jobName    = "test-job"
+		ns         = "default"
+	)
+
+	numJobsMatchTestCases := []struct {
```
Comment (dejanzele) [inline pkg/controllers/success_policy_test.go:142]:
```suggestion
	tests := []struct {
```

File: pkg/controllers/success_policy_test.go [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+	}
+}
+
+func TestNumJobsExpectedToSucceed(t *testing.T) {
+	var (
+		jobSetName = "test-jobset"
+		ns         = "default"
+	)
+
+	numJobsMatchTests := []struct {
```
Comment (dejanzele) [inline pkg/controllers/success_policy_test.go:230]:
```suggestion
	tests := []struct {
```

**Comment by zhifei92** [comment]
@dejanzele fix again，thank you

**Comment by kannon92** [comment]
Could you remove this line?

https://github.com/kubernetes-sigs/jobset/blob/main/pkg/controllers/success_policy.go#L24

**Comment by kannon92** [comment]
Once you address https://github.com/kubernetes-sigs/jobset/pull/501#issuecomment-2047518546, LGTM.

/assign @danielvegamyhre

File: pkg/controllers/success_policy_test.go [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+					JobLabels(map[string]string{jobset.ReplicatedJobNameKey: "test-replicated-job-1"}).
+					Obj(),
+				testutils.MakeJob(jobName, ns).
+					JobLabels(map[string]string{jobset.ReplicatedJobNameKey: "test-replicated-job-2"}).
+					Obj(),
+			},
+			expected: 2,
+		},
+		{
+			name: "non job matches the success policy",
```
Comment (danielvegamyhre) [inline pkg/controllers/success_policy_test.go:181]:
```suggestion
			name: "no jobs match the success policy",
```

File: pkg/controllers/success_policy_test.go [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+		ns         = "default"
+	)
+
+	tests := []struct {
+		name     string
+		js       *jobset.JobSet
+		expected int
+	}{
+		{
+			name: "any single job matching the jobSelector",
```
Comment (danielvegamyhre) [inline pkg/controllers/success_policy_test.go:236]:
```suggestion
			name: "any job completion fulfills success policy",
```

File: pkg/controllers/success_policy_test.go [hunk 7]
```diff
@@ -0,0 +1,283 @@
...
+			js: testutils.MakeJobSet(jobSetName, ns).
+				SuccessPolicy(&jobset.SuccessPolicy{
+					Operator: jobset.OperatorAny,
+				}).Obj(),
+			expected: 1,
+		},
+		{
+			name: "all replicated jobs match success policy",
```
Comment (danielvegamyhre) [inline pkg/controllers/success_policy_test.go:244]: In this test case, `"test-replicated-job-3"` does not match the success policy, so we should remove it so we are testing the case where all replicatedJobs match the success policy.

**Comment by zhifei92** [comment]
> Could you remove this line?
> 
> https://github.com/kubernetes-sigs/jobset/blob/main/pkg/controllers/success_policy.go#L24

done

**Comment by danielvegamyhre** [comment]
/lgtm
/approve

Thanks for working on this!

**Pull request merged**